src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -18,6 +18,34 @@ public partial class MemoryMappedFile
             HandleInheritability inheritability, MemoryMappedFileAccess access,
             MemoryMappedFileOptions options, long capacity)
         {
+            bool isSpecial = false;
+            if (fileStream != null)
+            {
+                if (fileStream.Length == 0)
```

Comment:
Getting the `Length` property of a `FileStream` will also call `FStat()`.  `stat()` isn't the fastest syscall in some file systems.

Reply:
good point. I did not realize that. There is no point in saving `fstat()` with `fstat`. 
I reworked the code to always call fstat() if the stream is not null and then use that for both checks instead of using fileStream.Length as we did before.